src/include/catalog/database_handle.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include <loggers/catalog_logger.h>
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "catalog/catalog.h"
+#include "catalog/namespace_handle.h"
+#include "storage/sql_table.h"
+#include "transaction/transaction_context.h"
+namespace terrier::catalog {
+
+class Catalog;
+/**
+ * A database handle represents a database in the system. It's the entry point for access data
+ * stored in this database.
+ */
+class DatabaseHandle {
+ public:
+  /**
+   * A database entry represent a row in pg_database catalog.
+   */
+  class DatabaseEntry {
+   public:
+    /**
+     * Constructs a database entry.
+     * @param oid the db_oid of the underlying database
+     * @param row a pointer points to the projection of the row
+     * @param map a map that encodes how to access attributes of the row
+     */
+    DatabaseEntry(db_oid_t oid, storage::ProjectedRow *row, storage::ProjectionMap map)
+        : oid_(oid), row_(row), map_(std::move(map)) {}
+    /**
+     * Get the value of an attribute
+     * @param col the col_oid of the attribute
+     * @return a pointer to the attribute value
+     */
+    byte *GetValue(col_oid_t col) { return row_->AccessWithNullCheck(map_[col]); }
```

Comment:
map_[col] may not be the desired behavior. If we access a non-existent column, it will silently insert and return physical offset 0, which in turn maps to ?? (currently the oid, but we sort so it depends what columns there are).

In other words, GetValue(nonexistent column) will always return the smallest column. I have changed this to .at() which errors out.